the-collab-lab/tcl-77-smart-shopping-list
# Issue #13: Sort item list by when to buy urgency

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description



Adds a text indicator of the buying urgency status of an item to the right of the item and sorts the list based on the status. If an item is inactive it will be at the bottom of the list, overdue items will be at the top, and if any items have the same days until next purchase they will be additionally sorted alphabetically.

## Related Issue


Closes #13 

## Acceptance Criteria



## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓    | 💯 Enhancement |
|  ✓    | ✨ New Feature |

## Updates

### Before


![Screenshot 2024-09-19 at 12 21 14 AM](https://github.com/user-attachments/assets/b83ea45b-4262-46a8-9f6c-cabd3179a3cf)


### After


![Screenshot 2024-09-19 at 12 21 39 AM](https://github.com/user-attachments/assets/0c3269d9-24fd-40e6-8609-17eeb4c691a1)


## Testing Steps / QA Criteria


- Click the [preview link](https://tcl-77-smart-shopping-list--pr35-bb-fz-sort-by-purcha-99ypueis.web.app/) for this PR
- Sign in with your account
- Click a list with item and navigate using the nav bar
- Items already on list should be sorted based on urgency status
- To see other status messages go to the [firebase](https://console.firebase.google.com/u/2/project/tcl-77-smart-shopping-list/firestore/data/~2F)
- Open the list and item you want to change
- Change to various dates to see status change
